diff --git a/node_modules/@graphql-mesh/json-schema/index.cjs.js b/node_modules/@graphql-mesh/json-schema/index.cjs.js
index f357434..0db07eb 100644
--- a/node_modules/@graphql-mesh/json-schema/index.cjs.js
+++ b/node_modules/@graphql-mesh/json-schema/index.cjs.js
@@ -620,6 +620,7 @@ class JsonSchemaHandler {
         };
     }
     async isGeneratedJSONSchemaValid({ samplePath, schemaPath }) {
+        if (this.parseString(samplePath)) return false;
         if (schemaPath || (!utils.isUrl(schemaPath) && (await utils.pathExists(schemaPath)))) {
             const [schemaFileStat, sampleFileStat] = await Promise.all([stat(schemaPath), stat(samplePath)]);
             if (schemaFileStat.mtime > sampleFileStat.mtime) {
@@ -633,7 +634,7 @@ class JsonSchemaHandler {
             if (samplePath in this.introspectionCache.schemaCache) {
                 return this.introspectionCache.schemaCache[samplePath];
             }
-            const sample = await utils.readFileOrUrlWithCache(samplePath, this.cache, { cwd: this.baseDir });
+            const sample = await this.getSample(samplePath);
             const schema = toJsonSchema(sample, {
                 required: false,
                 objects: {
@@ -656,6 +657,27 @@ class JsonSchemaHandler {
         }
         return null;
     }
+
+    async getSample(samplePath) {
+        let sample = this.parseString(samplePath);
+        if (!sample) {
+            sample = await utils.readFileOrUrlWithCache(samplePath, this.cache, {
+                cwd: this.baseDir,
+                headers: this.config.schemaHeaders
+            });
+        }
+        return sample;
+    }
+
+    parseString(jsonString) {
+        let data;
+        try {
+            data = JSON.parse(jsonString);
+        } catch (error) {
+            data = null;
+        }
+        return data;
+    }
 }
 
 module.exports = JsonSchemaHandler;
diff --git a/node_modules/@graphql-mesh/json-schema/index.esm.js b/node_modules/@graphql-mesh/json-schema/index.esm.js
index 7b53653..64a83e2 100644
--- a/node_modules/@graphql-mesh/json-schema/index.esm.js
+++ b/node_modules/@graphql-mesh/json-schema/index.esm.js
@@ -616,6 +616,7 @@ class JsonSchemaHandler {
         };
     }
     async isGeneratedJSONSchemaValid({ samplePath, schemaPath }) {
+        if (this.parseString(samplePath)) return false;
         if (schemaPath || (!isUrl(schemaPath) && (await pathExists(schemaPath)))) {
             const [schemaFileStat, sampleFileStat] = await Promise.all([stat(schemaPath), stat(samplePath)]);
             if (schemaFileStat.mtime > sampleFileStat.mtime) {
@@ -629,7 +630,7 @@ class JsonSchemaHandler {
             if (samplePath in this.introspectionCache.schemaCache) {
                 return this.introspectionCache.schemaCache[samplePath];
             }
-            const sample = await readFileOrUrlWithCache(samplePath, this.cache, { cwd: this.baseDir });
+            const sample = await this.getSample(samplePath);
             const schema = toJsonSchema(sample, {
                 required: false,
                 objects: {
@@ -652,6 +653,27 @@ class JsonSchemaHandler {
         }
         return null;
     }
+
+    async getSample(samplePath) {
+        let sample = this.parseString(samplePath);
+        if (!sample) {
+            sample = await utils.readFileOrUrlWithCache(samplePath, this.cache, {
+                cwd: this.baseDir,
+                headers: this.config.schemaHeaders
+            });
+        }
+        return sample;
+    }
+
+    parseString(jsonString) {
+        let data;
+        try {
+            data = JSON.parse(jsonString);
+        } catch (error) {
+            data = null;
+        }
+        return data;
+    }
 }
 
 export default JsonSchemaHandler;
