diff --git a/node_modules/@graphql-mesh/config/index.cjs.js b/node_modules/@graphql-mesh/config/index.cjs.js
index dcf4103..6906289 100644
--- a/node_modules/@graphql-mesh/config/index.cjs.js
+++ b/node_modules/@graphql-mesh/config/index.cjs.js
@@ -51,6 +51,7 @@ async function getPackage(name, type, importFn) {
         `${casedName}-${casedType}`,
         `${casedType}-${casedName}`,
         casedType,
+        name,
     ];
     if (name.includes('-')) {
         possibleNames.push(name);
